internal/pkg/deploy/cloudformation/project.go

```diff
@@ -95,7 +95,8 @@ func (cf CloudFormation) DelegateDNSPermissions(project *archer.Project, account
 	deployProject.DNSDelegationAccounts = append(dnsDelegatedAccounts, accountID)
 	updatedProjectConfig := stack.NewProjectStackConfig(&deployProject)
 
-	if err := cf.update(updatedProjectConfig); err != nil {
+	// swallow the errChangeSetEmpty error since it just means there were no updates needed.
+	if err := cf.update(updatedProjectConfig); err != nil && err != errChangeSetEmpty {
```

Comment:
Should we return `nil` when `err == errChangeSetEmpty` instead of calling l.103 `WaitUntilStackUpdateCompleteWithContext`?

Reply:
+1  - I think the behavior, otherwise, is that `WaitUntilStackUpdateCompleteWithContext` might wait forever or until the next change happens.